index.html

```diff
@@ -801,8 +801,14 @@ <h4 id="query-method">
                 </li>
               </ol>
             </li>
-            <li>If |permissionDesc|'s {{PermissionDescriptor/name}} member is not supported, return
-            [=a promise rejected with=] a {{TypeError}}.
+            <li>Let |rootDesc| be the object |permissionDesc| refers to, <a>converted to an IDL
+            value</a> of type {{PermissionDescriptor}}.
+            </li>
+            <li>If the conversion [=exception/throws=] an [=exception=], return <a>a promise
+            rejected with</a> that exception.
+            </li>
+            <li>If |rootDesc|'s {{PermissionDescriptor/name}} member is not supported, return [=a
```

Comment:
Nit (although I guess this might apply throughout in which case it's probably best done separately):
```suggestion
            <li>If |rootDesc|["{{PermissionDescriptor/name}}"] is not supported, return [=a
```